test/uk/gov/hmrc/tai/connectors/HttpHandlerSpec.scala

```diff
@@ -46,286 +59,281 @@ class HttpHandlerSpec extends BaseSpec with BeforeAndAfterEach {
+      result.responseCode mustBe BAD_REQUEST
+    }
+
+    "return Http exception for NOT_FOUND response" in {
+
+      server.stubFor(
+        post(anyUrl())
+          .willReturn(aResponse().withStatus(NOT_FOUND)))
+
+      val result = the[NotFoundException] thrownBy Await
+        .result(httpHandler.postToApi[String](testUrl, userInput), 5 seconds)
+
+      result.responseCode mustBe NOT_FOUND
 
-        result.responseCode mustBe GATEWAY_TIMEOUT
-      }
     }
+
   }
 
   "deleteFromApi" must {
-    "post request to DELETE and return the http response" when {
-      "http DELETE returns OK" in {
+    val userInput = "userInput"
 
-        when(http.DELETE[HttpResponse](any(), any())(any(), any(), any()))
-          .thenReturn(Future.successful(HttpResponse(OK)))
-        val result = Await.result(sut.deleteFromApi(mockUrl), 5 seconds)
-        result.status mustBe OK
-      }
+    List(
+      ACCEPTED,
+      OK,
+      NO_CONTENT
+    ).foreach { httpStatus =>
+      s"return json which is coming from http post call for $httpStatus response" in {
 
-      "http DELETE returns NO_CONTENT" in {
+        server.stubFor(
+          delete(anyUrl())
+            .willReturn(aResponse().withStatus(httpStatus)))
 
-        when(http.DELETE[HttpResponse](any(), any())(any(), any(), any()))
-          .thenReturn(Future.successful(HttpResponse(NO_CONTENT)))
-        val result = Await.result(sut.deleteFromApi(mockUrl), 5 seconds)
-        result.status mustBe NO_CONTENT
-      }
+        val result = Await.result(httpHandler.deleteFromApi(testUrl), 5 seconds)
+        result.status mustBe httpStatus
 
-      "http DELETE returns ACCEPTED" in {
-        when(http.DELETE[HttpResponse](any(), any())(any(), any(), any()))
-          .thenReturn(Future.successful(HttpResponse(ACCEPTED)))
-        val result = Await.result(sut.deleteFromApi(mockUrl), 5.seconds)
-        result.status mustBe ACCEPTED
       }
-
     }
-    "return Http exception" when {
-      "http response is NOT OK" in {
-        when(http.DELETE[HttpResponse](any(), any())(any(), any(), any()))
-          .thenReturn(Future.successful(HttpResponse(GATEWAY_TIMEOUT)))
 
-        val result = the[HttpException] thrownBy Await.result(sut.deleteFromApi(mockUrl), 5 seconds)
+    List(
+      GATEWAY_TIMEOUT,
+      INTERNAL_SERVER_ERROR,
+      SERVICE_UNAVAILABLE
+    ).foreach { httpStatus =>
+      s"return Http exception for $httpStatus response" in {
+
+        server.stubFor(
+          delete(anyUrl())
+            .willReturn(aResponse().withStatus(httpStatus).withBody("error response")))
+
+        val responseFuture = httpHandler.deleteFromApi(testUrl)
+        val ex = the[Upstream5xxResponse] thrownBy Await.result(responseFuture, 5 seconds)
+        ex.message must include("error response")
 
-        result.responseCode mustBe GATEWAY_TIMEOUT
       }
     }
-  }
 
-  private val mockUrl = "mockUrl"
+    "return Http exception for BAD_REQUEST response" in {
```

Comment:
This is throwing a BadRequestException, we should be as specific as we can

Reply:
changed the name to BadRequestException